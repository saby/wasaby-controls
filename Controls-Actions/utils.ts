import { IContextValue } from 'Controls/context';
import { Record } from 'Types/entity';

export interface IContextData extends IContextValue {
    documentContext?: Record;
}

/**
 * Получает объект из контекста
 * @public
 * @param context
 * @param objectName
 */
function findObject(context: IContextData, objectName: string): unknown {
    return context?.[objectName];
}

export { findObject };
