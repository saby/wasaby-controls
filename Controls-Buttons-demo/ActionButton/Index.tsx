import ButtonType from 'Controls-Buttons-meta/buttonConnectedButtonType';
import { Button } from 'Controls-Buttons/buttonConnected';
import { PropsDemoEditor } from 'Controls-demo/ObjectTypeEditor/PropsDemoEditor';
import { Provider as DataContextProvider } from 'Controls-DataEnv/context';
import { RecordSet } from 'Types/collection';
import { actions } from './actions';

function ButtonPropsEditor() {
    const contextConfig = {
        Actions: {
            dataFactoryName: 'Controls/dataFactory:List',
            dataFactoryArguments: {
                keyProperty: 'id',
                displayProperty: 'commandName',
                groupProperty: 'category',
            },
        },
    };

    const contextData = {
        Actions: {
            items: new RecordSet({
                keyProperty: 'id',
                rawData: actions,
            }),
        },
    };
    return (
        <DataContextProvider configs={contextConfig} loadResults={contextData}>
            <PropsDemoEditor metaType={ButtonType} control={Button} />;
        </DataContextProvider>
    );
}

export default ButtonPropsEditor;
