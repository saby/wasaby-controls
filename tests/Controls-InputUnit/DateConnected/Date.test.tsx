/**
 * @jest-environment jsdom
 */
import { unmountComponentAtNode } from 'react-dom';
import { WasabyEvents } from 'UICore/Events';
import { render } from '@testing-library/react';
import { getContent } from '../resources/utils';
import { default as DefaultValue } from 'Controls-Input-demo/DateConnected/Date/DefaultValue';
import { default as Base } from 'Controls-Input-demo/DateConnected/Date/Index';
import { default as Label } from 'Controls-Input-demo/DateConnected/Date/Label';
import { Controller } from 'Controls/popupTemplateStrategy';

describe(
    'Controls-Input/DateConnected:Date',
    () => {
        setTestID(148289607);

        let container = null;
        new Controller().init();

        beforeEach(() => {
            jest.useFakeTimers('modern');
            jest.setSystemTime(new Date(1996, 5, 21));
            container = document.createElement('div');
            WasabyEvents.initInstance(container);
            document.body.appendChild(container);
        });
        afterEach(() => {
            unmountComponentAtNode(container);
            WasabyEvents.destroyInstance(container);
            jest.useRealTimers();
            container.remove();
            container = null;
        });

        it('defaultValue', async () => {
            render(getContent(DefaultValue), {container});
            expect(container).toMatchSnapshot();
        });
        it('base', async () => {
            render(getContent(Base), {container});
            expect(container).toMatchSnapshot();
        });
        it('label', async () => {
            render(getContent(Label), {container});
            expect(container).toMatchSnapshot();
        });
    }
);
