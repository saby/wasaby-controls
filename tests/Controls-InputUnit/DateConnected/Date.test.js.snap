// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls-Input/DateConnected:Date base 1`] = `
<div>
  <div
    class="controlsDemo__wrapper controlsDemo_fixedWidth400 tw-flex tw-flex-col"
  >
    <div
      class="tw-flex controls-max-w-full tw-flex-col undefined controls-Input-connected_filled"
    >
      <a
        class="controls-Label controls-Label_underline_none tw-min-w-0"
      >
        <div
          class="controls-Label__wrapper"
        >
          <div
            class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
          >
            Mask DD.MM.YYYY
          </div>
          
        </div>
      </a>
      <div
        class="controls-Input-DatePicker"
        data-qa="controls-Input-DatePicker"
        tabindex="0"
      >
        <div
          class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Mask controls_Mask_theme-default controls-Input-DatePicker__content controls-DateBase controls-ValidateContainer__validationStatus-valid"
          data-qa="controls-BaseInput-Date"
          tabindex="0"
        >
          <div
            class="controls-Render__wrapper"
          >
            <span
              class="controls-Render__baseline"
            >
              ﻿
            </span>
            <div
              class="controls-InputBase__field controls-InputBase__field_autoWidth controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
              data-qa="controls-Render__field"
              tabindex="0"
            >
              <div
                class="controls-InputBase__stretcher"
              >
                <input
                  autocapitalize="off"
                  autocomplete="off"
                  autocorrect="off"
                  class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default  controls-InputBase__nativeField_stretcher controls-InputBase__stretcher-content"
                  inputmode="text"
                  name="ws-input_1996-06-20"
                  spellcheck="false"
                  tabindex="0"
                  type="text"
                  value="02.02.2022"
                />
                <div
                  class="controls-BaseInput__stretcher__container"
                >
                  <div
                    class="controls-InputBase__stretcher-block"
                  >
                    21.06.1996
                  </div>
                  <div
                    class="controls-InputBase__stretcher-block controls-BaseInput__stretcher-block"
                  >
                    Начало
                  </div>
                </div>
              </div>
              <div
                class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
              >
                <div
                  class="controls-Render__placeholder controls-Render__placeholder_overflow"
                >
                  Начало
                </div>
              </div>
            </div>
          </div>
        </div>
        <span
          class="controls-icon_size-s controls-icon_style-label icon-Calendar controls-icon controls-Input-DatePicker__button"
        />
      </div>
    </div>
    <div
      class="tw-flex controls-max-w-full tw-flex-col undefined controls-Input-connected_filled"
    >
      <a
        class="controls-Label controls-Label_underline_none tw-min-w-0"
      >
        <div
          class="controls-Label__wrapper"
        >
          <div
            class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
          >
            Mask DD.MM.YY
          </div>
          
        </div>
      </a>
      <div
        class="controls-Input-DatePicker"
        data-qa="controls-Input-DatePicker"
        tabindex="0"
      >
        <div
          class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Mask controls_Mask_theme-default controls-Input-DatePicker__content controls-DateBase controls-ValidateContainer__validationStatus-valid"
          data-qa="controls-BaseInput-Date"
          tabindex="0"
        >
          <div
            class="controls-Render__wrapper"
          >
            <span
              class="controls-Render__baseline"
            >
              ﻿
            </span>
            <div
              class="controls-InputBase__field controls-InputBase__field_autoWidth controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
              data-qa="controls-Render__field"
              tabindex="0"
            >
              <div
                class="controls-InputBase__stretcher"
              >
                <input
                  autocapitalize="off"
                  autocomplete="off"
                  autocorrect="off"
                  class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default  controls-InputBase__nativeField_stretcher controls-InputBase__stretcher-content"
                  inputmode="text"
                  name="ws-input_1996-06-20"
                  spellcheck="false"
                  tabindex="0"
                  type="text"
                  value="02.02.22"
                />
                <div
                  class="controls-BaseInput__stretcher__container"
                >
                  <div
                    class="controls-InputBase__stretcher-block"
                  >
                    21.06.96
                  </div>
                  <div
                    class="controls-InputBase__stretcher-block controls-BaseInput__stretcher-block"
                  >
                    Начало
                  </div>
                </div>
              </div>
              <div
                class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
              >
                <div
                  class="controls-Render__placeholder controls-Render__placeholder_overflow"
                >
                  Начало
                </div>
              </div>
            </div>
          </div>
        </div>
        <span
          class="controls-icon_size-s controls-icon_style-label icon-Calendar controls-icon controls-Input-DatePicker__button"
        />
      </div>
    </div>
    <div
      class="tw-flex controls-max-w-full tw-flex-col undefined controls-Input-connected_filled"
    >
      <a
        class="controls-Label controls-Label_underline_none tw-min-w-0"
      >
        <div
          class="controls-Label__wrapper"
        >
          <div
            class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
          >
            Mask MM.YYYY
          </div>
          
        </div>
      </a>
      <div
        class="controls-Input-DatePicker"
        data-qa="controls-Input-DatePicker"
        tabindex="0"
      >
        <div
          class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Mask controls_Mask_theme-default controls-Input-DatePicker__content controls-DateBase controls-ValidateContainer__validationStatus-valid"
          data-qa="controls-BaseInput-Date"
          tabindex="0"
        >
          <div
            class="controls-Render__wrapper"
          >
            <span
              class="controls-Render__baseline"
            >
              ﻿
            </span>
            <div
              class="controls-InputBase__field controls-InputBase__field_autoWidth controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
              data-qa="controls-Render__field"
              tabindex="0"
            >
              <div
                class="controls-InputBase__stretcher"
              >
                <input
                  autocapitalize="off"
                  autocomplete="off"
                  autocorrect="off"
                  class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default  controls-InputBase__nativeField_stretcher controls-InputBase__stretcher-content"
                  inputmode="text"
                  name="ws-input_1996-06-20"
                  spellcheck="false"
                  tabindex="0"
                  type="text"
                  value="02.2022"
                />
                <div
                  class="controls-BaseInput__stretcher__container"
                >
                  <div
                    class="controls-InputBase__stretcher-block"
                  >
                    06.1996
                  </div>
                  <div
                    class="controls-InputBase__stretcher-block controls-BaseInput__stretcher-block"
                  >
                    Начало
                  </div>
                </div>
              </div>
              <div
                class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
              >
                <div
                  class="controls-Render__placeholder controls-Render__placeholder_overflow"
                >
                  Начало
                </div>
              </div>
            </div>
          </div>
        </div>
        <span
          class="controls-icon_size-s controls-icon_style-label icon-Calendar controls-icon controls-Input-DatePicker__button"
        />
      </div>
    </div>
    <div
      class="tw-flex controls-max-w-full tw-flex-col undefined controls-Input-connected_filled"
    >
      <a
        class="controls-Label controls-Label_underline_none tw-min-w-0"
      >
        <div
          class="controls-Label__wrapper"
        >
          <div
            class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
          >
            Mask YYYY
          </div>
          
        </div>
      </a>
      <div
        class="controls-Input-DatePicker"
        data-qa="controls-Input-DatePicker"
        tabindex="0"
      >
        <div
          class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Mask controls_Mask_theme-default controls-Input-DatePicker__content controls-DateBase controls-ValidateContainer__validationStatus-valid"
          data-qa="controls-BaseInput-Date"
          tabindex="0"
        >
          <div
            class="controls-Render__wrapper"
          >
            <span
              class="controls-Render__baseline"
            >
              ﻿
            </span>
            <div
              class="controls-InputBase__field controls-InputBase__field_autoWidth controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
              data-qa="controls-Render__field"
              tabindex="0"
            >
              <div
                class="controls-InputBase__stretcher"
              >
                <input
                  autocapitalize="off"
                  autocomplete="off"
                  autocorrect="off"
                  class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default  controls-InputBase__nativeField_stretcher controls-InputBase__stretcher-content"
                  inputmode="text"
                  name="ws-input_1996-06-20"
                  spellcheck="false"
                  tabindex="0"
                  type="text"
                  value="2022"
                />
                <div
                  class="controls-BaseInput__stretcher__container"
                >
                  <div
                    class="controls-InputBase__stretcher-block"
                  >
                    1996
                  </div>
                  <div
                    class="controls-InputBase__stretcher-block controls-BaseInput__stretcher-block"
                  >
                    Начало
                  </div>
                </div>
              </div>
              <div
                class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
              >
                <div
                  class="controls-Render__placeholder controls-Render__placeholder_overflow"
                >
                  Начало
                </div>
              </div>
            </div>
          </div>
        </div>
        <span
          class="controls-icon_size-s controls-icon_style-label icon-Calendar controls-icon controls-Input-DatePicker__button"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Controls-Input/DateConnected:Date defaultValue 1`] = `
<div>
  <div
    class="controlsDemo__wrapper controlsDemo_fixedWidth400 tw-flex tw-flex-col"
  >
    <div
      class="tw-flex controls-max-w-full tw-flex-col undefined controls-Input-connected_filled"
    >
      <a
        class="controls-Label controls-Label_underline_none tw-min-w-0"
      >
        <div
          class="controls-Label__wrapper"
        >
          <div
            class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
          >
            defaultValue
          </div>
          
        </div>
      </a>
      <div
        class="controls-Input-DatePicker"
        data-qa="controls-Input-DatePicker"
        tabindex="0"
      >
        <div
          class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Mask controls_Mask_theme-default controls-Input-DatePicker__content controls-DateBase controls-ValidateContainer__validationStatus-valid"
          data-qa="controls-BaseInput-Date"
          tabindex="0"
        >
          <div
            class="controls-Render__wrapper"
          >
            <span
              class="controls-Render__baseline"
            >
              ﻿
            </span>
            <div
              class="controls-InputBase__field controls-InputBase__field_autoWidth controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
              data-qa="controls-Render__field"
              tabindex="0"
            >
              <div
                class="controls-InputBase__stretcher"
              >
                <input
                  autocapitalize="off"
                  autocomplete="off"
                  autocorrect="off"
                  class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default  controls-InputBase__nativeField_stretcher controls-InputBase__stretcher-content"
                  inputmode="text"
                  name="ws-input_1996-06-20"
                  spellcheck="false"
                  tabindex="0"
                  type="text"
                  value="03.04.23"
                />
                <div
                  class="controls-BaseInput__stretcher__container"
                >
                  <div
                    class="controls-InputBase__stretcher-block"
                  >
                    21.06.96
                  </div>
                  <div
                    class="controls-InputBase__stretcher-block controls-BaseInput__stretcher-block"
                  >
                    Начало
                  </div>
                </div>
              </div>
              <div
                class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
              >
                <div
                  class="controls-Render__placeholder controls-Render__placeholder_overflow"
                >
                  Начало
                </div>
              </div>
            </div>
          </div>
        </div>
        <span
          class="controls-icon_size-s controls-icon_style-label icon-Calendar controls-icon controls-Input-DatePicker__button"
        />
      </div>
    </div>
    <div
      class="tw-flex controls-max-w-full tw-flex-col undefined "
    >
      <a
        class="controls-Label controls-Label_underline_none tw-min-w-0"
      >
        <div
          class="controls-Label__wrapper"
        >
          <div
            class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
          >
            defaultValue is empty
          </div>
          
        </div>
      </a>
      <div
        class="controls-Input-DatePicker"
        data-qa="controls-Input-DatePicker"
        tabindex="0"
      >
        <div
          class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-DateInput__placeholder controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Mask controls_Mask_theme-default controls-Input-DatePicker__content controls-DateBase controls-ValidateContainer__validationStatus-valid"
          data-qa="controls-BaseInput-Date"
          tabindex="0"
        >
          <div
            class="controls-Render__wrapper"
          >
            <span
              class="controls-Render__baseline"
            >
              ﻿
            </span>
            <div
              class="controls-InputBase__field controls-InputBase__field_autoWidth controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
              data-qa="controls-Render__field"
              tabindex="0"
            >
              <div
                class="controls-InputBase__stretcher"
              >
                <input
                  autocapitalize="off"
                  autocomplete="off"
                  autocorrect="off"
                  class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default  controls-InputBase__nativeField_stretcher controls-InputBase__stretcher-content"
                  inputmode="text"
                  name="ws-input_1996-06-20"
                  spellcheck="false"
                  tabindex="0"
                  type="text"
                  value="Начало"
                />
                <div
                  class="controls-BaseInput__stretcher__container"
                >
                  <div
                    class="controls-InputBase__stretcher-block"
                  >
                    21.06.96
                  </div>
                  <div
                    class="controls-InputBase__stretcher-block controls-BaseInput__stretcher-block"
                  >
                    Начало
                  </div>
                </div>
              </div>
              <div
                class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
              >
                <div
                  class="controls-Render__placeholder controls-Render__placeholder_overflow"
                >
                  Начало
                </div>
              </div>
            </div>
          </div>
        </div>
        <span
          class="controls-icon_size-s controls-icon_style-label icon-Calendar controls-icon controls-Input-DatePicker__button"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Controls-Input/DateConnected:Date label 1`] = `
<div>
  <div
    class="controlsDemo__wrapper controlsDemo_fixedWidth300 tw-flex tw-flex-col"
  >
    <div
      class="controls-Input-DatePicker controls-margin_top-m controls-margin_top-m controls-Input-connected_filled tw-w-full"
      data-qa="controls-Input-DatePicker"
      tabindex="0"
    >
      <div
        class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Mask controls_Mask_theme-default controls-Input-DatePicker__content controls-DateBase controls-ValidateContainer__validationStatus-valid"
        data-qa="controls-BaseInput-Date"
        tabindex="0"
      >
        <div
          class="controls-Render__wrapper"
        >
          <span
            class="controls-Render__baseline"
          >
            ﻿
          </span>
          <div
            class="controls-InputBase__field controls-InputBase__field_autoWidth controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
            data-qa="controls-Render__field"
            tabindex="0"
          >
            <div
              class="controls-InputBase__stretcher"
            >
              <input
                autocapitalize="off"
                autocomplete="off"
                autocorrect="off"
                class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default  controls-InputBase__nativeField_stretcher controls-InputBase__stretcher-content"
                inputmode="text"
                name="ws-input_1996-06-20"
                spellcheck="false"
                tabindex="0"
                type="text"
                value="02.02.22"
              />
              <div
                class="controls-BaseInput__stretcher__container"
              >
                <div
                  class="controls-InputBase__stretcher-block"
                >
                  21.06.96
                </div>
                <div
                  class="controls-InputBase__stretcher-block controls-BaseInput__stretcher-block"
                >
                  Начало
                </div>
              </div>
            </div>
            <div
              class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
            >
              <div
                class="controls-Render__placeholder controls-Render__placeholder_overflow"
              >
                Начало
              </div>
            </div>
          </div>
        </div>
      </div>
      <span
        class="controls-icon_size-s controls-icon_style-label icon-Calendar controls-icon controls-Input-DatePicker__button"
      />
    </div>
    <div
      class="tw-flex controls-max-w-full tw-flex-col controls-margin_top-m controls-Input-connected_filled"
    >
      <a
        class="controls-Label controls-Label_underline_none tw-min-w-0"
      >
        <div
          class="controls-Label__wrapper"
        >
          <div
            class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
          >
            label top
          </div>
          
        </div>
      </a>
      <div
        class="controls-Input-DatePicker controls-margin_top-m"
        data-qa="controls-Input-DatePicker"
        tabindex="0"
      >
        <div
          class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Mask controls_Mask_theme-default controls-Input-DatePicker__content controls-DateBase controls-ValidateContainer__validationStatus-valid"
          data-qa="controls-BaseInput-Date"
          tabindex="0"
        >
          <div
            class="controls-Render__wrapper"
          >
            <span
              class="controls-Render__baseline"
            >
              ﻿
            </span>
            <div
              class="controls-InputBase__field controls-InputBase__field_autoWidth controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
              data-qa="controls-Render__field"
              tabindex="0"
            >
              <div
                class="controls-InputBase__stretcher"
              >
                <input
                  autocapitalize="off"
                  autocomplete="off"
                  autocorrect="off"
                  class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default  controls-InputBase__nativeField_stretcher controls-InputBase__stretcher-content"
                  inputmode="text"
                  name="ws-input_1996-06-20"
                  spellcheck="false"
                  tabindex="0"
                  type="text"
                  value="02.02.22"
                />
                <div
                  class="controls-BaseInput__stretcher__container"
                >
                  <div
                    class="controls-InputBase__stretcher-block"
                  >
                    21.06.96
                  </div>
                  <div
                    class="controls-InputBase__stretcher-block controls-BaseInput__stretcher-block"
                  >
                    Начало
                  </div>
                </div>
              </div>
              <div
                class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
              >
                <div
                  class="controls-Render__placeholder controls-Render__placeholder_overflow"
                >
                  Начало
                </div>
              </div>
            </div>
          </div>
        </div>
        <span
          class="controls-icon_size-s controls-icon_style-label icon-Calendar controls-icon controls-Input-DatePicker__button"
        />
      </div>
    </div>
    <div
      class="tw-flex controls-max-w-full tw-flex-row tw-items-baseline controls-margin_top-m controls-Input-connected_filled"
    >
      <a
        class="controls-Label controls-Label_underline_none tw-min-w-0 tw-flex-shrink"
      >
        <div
          class="controls-Label__wrapper"
        >
          <div
            class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
          >
            label start
          </div>
          
        </div>
      </a>
      <div
        class="controls-Input-DatePicker controls-margin_top-m"
        data-qa="controls-Input-DatePicker"
        tabindex="0"
      >
        <div
          class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Mask controls_Mask_theme-default controls-Input-DatePicker__content controls-DateBase controls-ValidateContainer__validationStatus-valid"
          data-qa="controls-BaseInput-Date"
          tabindex="0"
        >
          <div
            class="controls-Render__wrapper"
          >
            <span
              class="controls-Render__baseline"
            >
              ﻿
            </span>
            <div
              class="controls-InputBase__field controls-InputBase__field_autoWidth controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
              data-qa="controls-Render__field"
              tabindex="0"
            >
              <div
                class="controls-InputBase__stretcher"
              >
                <input
                  autocapitalize="off"
                  autocomplete="off"
                  autocorrect="off"
                  class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default  controls-InputBase__nativeField_stretcher controls-InputBase__stretcher-content"
                  inputmode="text"
                  name="ws-input_1996-06-20"
                  spellcheck="false"
                  tabindex="0"
                  type="text"
                  value="02.02.22"
                />
                <div
                  class="controls-BaseInput__stretcher__container"
                >
                  <div
                    class="controls-InputBase__stretcher-block"
                  >
                    21.06.96
                  </div>
                  <div
                    class="controls-InputBase__stretcher-block controls-BaseInput__stretcher-block"
                  >
                    Начало
                  </div>
                </div>
              </div>
              <div
                class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
              >
                <div
                  class="controls-Render__placeholder controls-Render__placeholder_overflow"
                >
                  Начало
                </div>
              </div>
            </div>
          </div>
        </div>
        <span
          class="controls-icon_size-s controls-icon_style-label icon-Calendar controls-icon controls-Input-DatePicker__button"
        />
      </div>
    </div>
    <div
      class="tw-flex controls-max-w-full tw-flex-row tw-items-baseline controls-margin_top-m controls-Input-connected_filled"
    >
      <span
        class="controls-icon_size-s controls-icon_style-secondary icon-SabyBird controls-icon controls-margin_right-s"
      />
      <div
        class="controls-Input-DatePicker controls-margin_top-m"
        data-qa="controls-Input-DatePicker"
        tabindex="0"
      >
        <div
          class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Mask controls_Mask_theme-default controls-Input-DatePicker__content controls-DateBase controls-ValidateContainer__validationStatus-valid"
          data-qa="controls-BaseInput-Date"
          tabindex="0"
        >
          <div
            class="controls-Render__wrapper"
          >
            <span
              class="controls-Render__baseline"
            >
              ﻿
            </span>
            <div
              class="controls-InputBase__field controls-InputBase__field_autoWidth controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
              data-qa="controls-Render__field"
              tabindex="0"
            >
              <div
                class="controls-InputBase__stretcher"
              >
                <input
                  autocapitalize="off"
                  autocomplete="off"
                  autocorrect="off"
                  class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default  controls-InputBase__nativeField_stretcher controls-InputBase__stretcher-content"
                  inputmode="text"
                  name="ws-input_1996-06-20"
                  spellcheck="false"
                  tabindex="0"
                  type="text"
                  value="02.02.22"
                />
                <div
                  class="controls-BaseInput__stretcher__container"
                >
                  <div
                    class="controls-InputBase__stretcher-block"
                  >
                    21.06.96
                  </div>
                  <div
                    class="controls-InputBase__stretcher-block controls-BaseInput__stretcher-block"
                  >
                    Начало
                  </div>
                </div>
              </div>
              <div
                class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
              >
                <div
                  class="controls-Render__placeholder controls-Render__placeholder_overflow"
                >
                  Начало
                </div>
              </div>
            </div>
          </div>
        </div>
        <span
          class="controls-icon_size-s controls-icon_style-label icon-Calendar controls-icon controls-Input-DatePicker__button"
        />
      </div>
    </div>
  </div>
</div>
`;
